### PR TITLE
[RTL] Add extra argument to `remove_paragraph` to skip cache invalidation and a method for manual cache invalidation.

### PR DESCRIPTION
**Pull request by bruvzg**
Required to quickly remove paragraph for the beginning of log without full refresh, see https://github.com/godotengine/godot/pull/91012

### CODE REVIEW

File: misc/extension_api_validation/4.2-stable.expected [hunk 1]
```diff
@@ -315,3 +315,10 @@ Validate extension JSON: Error: Field 'classes/TextServer/methods/shaped_text_ge
 Validate extension JSON: Error: Field 'classes/TextServerExtension/methods/_shaped_text_get_word_breaks/arguments': size changed value in new API, from 2 to 3.
 
 Added optional argument. Compatibility method registered.
+
+
+GH-91098
+--------
+Validate extension JSON: Error: Field 'classes/RichTextLabel/methods/remove_paragraph/arguments': size changed value in new API, from 1 to 2.
+
+Added optional argument. Compatibility method registered.
```
Comment (akien-mga) [inline misc/extension_api_validation/4.2-stable.expected:324]: Missing newline at eof.

**Review by akien-mga — APPROVED** [review]

**Review by AThousandShips — APPROVED** [review]

**Comment by akien-mga** [comment]
Needs a rebase and some style fixes.

**Comment by akien-mga** [comment]
Thanks!

**Comment by TheSofox** [comment]
Fantastic work @bruvzg, I really appreciate all the work you've put into this.

**Pull request merged**